sys/androidmedia/gstamc.c

```diff
@@ -1556,10 +1604,16 @@ scan_codecs (GstPlugin * plugin)
 
       for (j = 0; j < gst_codec_info->n_supported_types; j++) {
         GstAmcCodecType *gst_codec_type = &gst_codec_info->supported_types[j];
+        gint k;
 
         g_free (gst_codec_type->mime);
         g_free (gst_codec_type->color_formats);
         g_free (gst_codec_type->profile_levels);
+
+        for (k = 0; k < gst_codec_type->n_features; k++) {
+          g_free (&(gst_codec_type->features[k]));
```

Comment:
This code block is wrong and will lead to a crash.
gst_codec_type->features is allocated as one chunk of memory, not as several ones.
it should be one

g_free (gst_codec_type->features);

instead of loop